tests/components/lutron_caseta/test_config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Test the Lutron Caseta config flow."""
+from asynctest import patch
+from pylutron_caseta.smartbridge import Smartbridge
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.lutron_caseta import DOMAIN
+import homeassistant.components.lutron_caseta.config_flow as CasetaConfigFlow
+from homeassistant.components.lutron_caseta.const import (
+    CONF_CA_CERTS,
+    CONF_CERTFILE,
+    CONF_KEYFILE,
+    ERROR_CANNOT_CONNECT,
+    STEP_IMPORT_FAILED,
+)
+from homeassistant.const import CONF_HOST
+
+from tests.common import MockConfigEntry
+
+
+class MockBridge:
+    """Mock Lutron bridge that emulates configured connected status."""
+
+    def __init__(self, can_connect=True):
+        """Initialize MockBridge instance with configured mock connectivity."""
+        self.can_connect = can_connect
+        self.is_currently_connected = False
+
+    async def connect(self):
+        """Connect the mock bridge."""
+        if self.can_connect:
+            self.is_currently_connected = True
+
+    def is_connected(self):
+        """Return whether the mock bridge is connected."""
+        return self.is_currently_connected
+
+    async def close(self):
+        """Close the mock bridge connection."""
+        self.is_currently_connected = False
+
+
+async def test_bridge_import_flow(hass):
+    """Test a bridge entry gets created and set up during the import flow."""
+
+    entry_mock_data = {
+        CONF_HOST: "1.1.1.1",
+        CONF_KEYFILE: "",
+        CONF_CERTFILE: "",
+        CONF_CA_CERTS: "",
+    }
+
+    with patch(
+        "homeassistant.components.lutron_caseta.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch.object(Smartbridge, "create_tls") as create_tls:
+        create_tls.return_value = MockBridge(can_connect=True)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=entry_mock_data,
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == CasetaConfigFlow.ENTRY_DEFAULT_TITLE
+    assert result["data"] == entry_mock_data
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)